client/ayon_core/plugins/publish/integrate_review.py

```diff
@@ -100,3 +104,55 @@ def _get_review_label(self, repre, uploaded_labels):
             idx += 1
             label = f"{orig_label}_{idx}"
         return label
+
+    def _upload_with_retries(
+        self,
+        ayon_con,
+        endpoint,
+        repre_path,
+        headers,
+        max_retries: int = 3,
+        backoff_seconds: int = 2,
+    ):
+        """Upload file with simple exponential backoff retries.
+
+        If all retries fail we raise a PublishXmlValidationError with a help key
+        to guide the user to retry publish.
+        """
+        last_error = None
+        for attempt in range(1, max_retries + 1):
+            try:
+                ayon_con.upload_file(
+                    endpoint,
+                    repre_path,
+                    headers=headers,
+                    request_type=RequestTypes.post,
+                )
+                return
+            except Exception as exc:  # noqa: BLE001 - bubble after retries
```

Comment:
Don't we know the exception?

Reply:
Agreed that we should not retry on “anything.” The intent was to catch transient upload failures only. I will narrow the except clause to network or server-side cases that are safe to retry and let everything else bubble immediately. Concretely something like:

from requests import exceptions as req_exc

retryable = (
    req_exc.Timeout,
    req_exc.ConnectionError,
)